google_compute_engine_oslogin/pam_module/pam_oslogin_login.cc

```diff
@@ -58,9 +58,9 @@ PAM_EXTERN int pam_sm_acct_mgmt(pam_handle_t *pamh, int flags, int argc,
   url << kMetadataServerUrl << "users?username=" << UrlEncode(str_user_name);
   string response;
   long http_code = 0;
-  if (!HttpGet(url.str(), &response, &http_code) || http_code == 500) {
-    // First check if there is a local file indicating that this is an oslogin
-    // user that has logged in before.
+  // Check for server failures. In this case, we can check for a local file to
+  // see if the user is an oslogin user.
+  if (!HttpGet(url.str(), &response, &http_code) || http_code >= 500) {
     if (file_exists) {
       return PAM_PERM_DENIED;
     }
```

Comment:
It is not true, the server now always returns 404 for non-existent users. The check for empty response is leftover from testing in development. We should explicitly be checking for 404 here.